### PR TITLE
StoreLocator - Cache flush to Varnish only for needed tag not FULL ca…

### PR DESCRIPTION
**Pull request by IgorSIT**
…che flush to Varnish.

### CODE REVIEW

**Comment by Fdec** [comment]
Hi @delyriand 

Have you test this fix ?

File: Observer/CleanStoreLocatorCache.php [hunk 1]
```diff
@@ -53,8 +61,13 @@ public function execute(Observer $observer)
...
         /** @var SellerInterface $seller */
         $seller = $observer->getEvent()->getSeller();
+
         if ($seller->hasDataChanges()) {
-            $this->cache->clean(\Zend_Cache::CLEANING_MODE_MATCHING_TAG, [Search::CACHE_TAG]);
+            $formattedTag = sprintf('((^|,)%s(,|$))', Search::CACHE_TAG);
+
+            $this->purgeCache->sendPurgeRequest($formattedTag);
+
+            $this->cache->clean([Search::CACHE_TAG]);
```
[the remark this answers is not in the document]
Comment (IgorSIT) [reply Observer/CleanStoreLocatorCache.php]: @delyriand Hi,
If use only second method 
`$this->cache->clean([Search::CACHE_TAG]) `
then after save retailer on BO and reload stores page (F5) not "Empty Cache And Hard Reload" on front, the stores data still cached and displays incorrect (old data) stores data on the page. It is not good for stores page.
But if use two methods then the stores data is correct (display new data) after reload stores page (F5).
The first method 
`$this->purgeCache->sendPurgeRequest($formattedTag)`
sends curl purge request to servers to invalidate Varnish cache by tags pattern (see vendor/magento/module-cache-invalidate/Model/PurgeCache::sendPurgeRequestToServers()).
So, I think that for correct cache flush to Varnish need use these two methods.

[the remark this answers is not in the document]
Comment (delyriand) [reply Observer/CleanStoreLocatorCache.php]: OK, replace `sendPurgeRequest` by a dispatch event 'clean_cache_by_tags` as seen together

[the remark this answers is not in the document]
Comment (delyriand) [reply Observer/CleanStoreLocatorCache.php]: and fix `Smile\StoreLocator\Block\Search` class for use the cache :)

[the remark this answers is not in the document]
Comment (IgorSIT) [reply Observer/CleanStoreLocatorCache.php]: @delyriand 
I updated my Pull Request.

**Pull request merged**